src/core/react/Canvas.tsx

```diff
@@ -42,98 +42,129 @@ export interface Control {
     reset(): void;
 }
 
-const CanvasComponent = React.memo((props: SizeMeProps & CanvasProps) => {
+const CanvasComponent = React.memo((props: CanvasProps) => {
     const {
         className,
         contentHeight,
         contentWidth,
         onRender,
         padding,
-        size,
     } = props;
 
+    const canvasRef = React.useRef<HTMLDivElement>(null);
+    const { width, height } = useResizeObserver(canvasRef);
+
     const [transform, setTransform] = React.useState<Transform>(DEFAULT_TRANSFORM);
     const contentHeightRef = React.useRef(contentHeight);
     const contentWidthRef = React.useRef(contentWidth);
     const selectionRef = React.useRef<Selection<HTMLDivElement, unknown, any, any>>();
-    const sizeRef = React.useRef(size);
-    const zoomRef = React.useRef<ZoomBehavior<Element, unknown>>();
+    const sizeRef = React.useRef({ width, height });
+    const zoomRef = React.useRef<ZoomBehavior<HTMLDivElement, unknown>>();
 
     contentWidthRef.current = contentWidth;
     contentHeightRef.current = contentHeight;
-    sizeRef.current = size;
-    
-    const doInit = React.useCallback((container: HTMLDivElement) => {
-        if (!container) {
+
+    React.useEffect(() => {
+        sizeRef.current = { width, height };
+    }, [width, height]);
+
+    React.useEffect(() => {
+        const element = canvasRef.current;
+        if (!element || zoomRef.current) {
             return;
         }
 
-        zoomRef.current = zoom()
+        const newZoomBehavior = zoom<HTMLDivElement, unknown>()
             .scaleExtent([0.5, 4])
             .filter(event => {
-                return event.button === 1 || event.type === 'wheel';
+                const e = event as WheelEvent | MouseEvent;
```

Comment:
I think you can assign the type directly in the event .filter((event: WheelEvent | MosueEvent))